src/lsm/set_associative_cache.zig

```diff
@@ -176,6 +177,16 @@ pub fn SetAssociativeCache(
             self.counts.set(set.offset + way, 0);
         }
 
+        /// Hint that the key is less likely to be accessed in the future, without actually removing
+        /// it from the cache.
+        pub fn demote(self: *Self, key: Key) void {
+            const set = self.associate(key);
+            const way = self.search(set, key) orelse return;
+
+            const count = self.counts.get(set.offset + way);
+            self.counts.set(set.offset + way, div_ceil(count, 2));
```

Comment:
I'll set the count to 1, but I'm going to leave the clock hand alone for predictability. (And I think it could actually be worse when a batch of blocks are demoted if we move the clock hand to the last one.)

Reply:
Spot on about not messing with the clock hand.